ocs_ci/deployment/metallb.py

```diff
@@ -110,39 +112,40 @@ def create_catalog_source(
 
         """
         logger.info("Creating catalog source for MetalLB")
-        # replace latest version with specific version
-        catalog_source_data = templating.load_yaml(QE_APP_REGISTRY_SOURCE)
-        if not metallb_version:
-            metallb_version = get_ocp_version()
-
-        image_placeholder = catalog_source_data.get("spec").get("image")
-        catalog_source_data.get("spec").update(
-            {"image": image_placeholder.format(metallb_version)}
-        )
-
-        self.catalog_source_name = catalog_source_data.get("metadata").get("name")
 
-        if self.catalog_source_created():
-            logger.info(f"Catalog Source {self.catalog_source_name} already exists")
-            return
+        self.catalog_source_name = self.get_catsrc_name()
 
-        # install catalog source
-        metallb_catalog_file = tempfile.NamedTemporaryFile(
-            mode="w+", prefix="metallb_catalogsource", delete=False
-        )
-        templating.dump_data_to_temp_yaml(
-            catalog_source_data, metallb_catalog_file.name
-        )
+        if not self.catalog_source_created():
+            self.correct_idms()
 
-        exec_cmd(f"oc apply -f {metallb_catalog_file.name}", timeout=2400)
+            create_optional_operators_catalogsource_non_ga(force=True)
+            metallb_catalog_source = CatalogSource(
+                resource_name=self.catalog_source_name,
+                namespace=MARKETPLACE_NAMESPACE,
+            )
+            return metallb_catalog_source.wait_for_state("READY")
+        else:
+            logger.info(
+                f"Catalog Source {constants.QE_APP_REGISTRY_CATALOG_SOURCE_NAME} already exists"
+            )
+            return True
 
-        # wait for catalog source is ready
-        metallb_catalog_source = CatalogSource(
-            resource_name=self.catalog_source_name,
-            namespace=MARKETPLACE_NAMESPACE,
+    def get_catsrc_name(self):
+        """
+        Helper function to get the catalog source name
```

Comment:
Docstring with return value should be included.

Reply:
fixed